### DISCUSSION TITLE
override/priority of classes based on their order

### DISCUSSION THREAD

**Discussion by MikeBusuttil** [tailwindlabs/tailwindcss#6795] [Ideas]
consider the following:
`<p class="text-gray-600 text-white">hi</p>`
`<p class="text-white text-gray-600">hi</p>`
Both will yield gray text instead of white.  It would be helpful as a developer if the order would determine the applied class.

**Comment by JadedBlueEyes** [comment]
This is impossible using CSS, and leads to bad code anyway.

If the element needs multiple different colours, selectors with more specificity will take precedence. For example,

```html
<p class="text-gray-600 hover:text-white">hi</p>
```

Will be grey, and white when hovered over, and

```html
<p class="text-white hover:text-gray-600">hi</p>
```

will be the other way around - white, and grey when hovered over.

**Reply by MikeBusuttil** [reply]
consider the case of defining a reusable component in a JavaScript framework where you want that component to have:
1. a default background color to be used when a background color isn't specified
2. the ability to override that background color on occasion by specifying a background color

**Reply by JadedBlueEyes** [reply]
In that case, you should be using Javascript to set the class, and can do something like `class={classNames||"default classes here"}`.

**Reply by MikeBusuttil** [reply]
we do, but its clunky and less elegant, which is why I've submitted this as a feature request

**Reply by adamwathan** [reply]
As @JoelEllis mentioned this isn't possible, the browser doesn't care what order your class names are in in your HTML, it only cares about the order in your CSS.
